### PR TITLE
DCS-1353:Create prison API endpoint to confirm arrival of a temporary absence

### PR DESCRIPTION
**Pull request by michalnawrockidj**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/service/TemporaryAbsenceArrivalServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Test
+    @Sql(scripts = {"/sql/temporaryAbsenceArrival_init.sql"},
+        executionPhase = ExecutionPhase.BEFORE_TEST_METHOD,
+        config = @SqlConfig(transactionMode = TransactionMode.ISOLATED))
+    @Sql(scripts = {"/sql/temporaryAbsenceArrival_clean.sql"},
+        executionPhase = ExecutionPhase.AFTER_TEST_METHOD,
+        config = @SqlConfig(transactionMode = TransactionMode.ISOLATED))
+    public void temporaryAbsenceArrival() {
+        RequestForTemporaryAbsenceArrival requestForCourtTransferIn = new RequestForTemporaryAbsenceArrival();
```
Comment (bell-pa) [inline src/test/java/uk/gov/justice/hmpps/prison/service/TemporaryAbsenceArrivalServiceTest.java]: Use a more appropriate name please.

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java [hunk 2]
```diff
@@ -848,6 +852,71 @@ public InmateDetail courtTransferIn(final String offenderNo, final RequestForCou
...
+
+        final LocalDateTime movementTime = getAndCheckMovementTime(requestForTemporaryAbsenceArrival.getDateTime(), offenderBooking.getBookingId());
+
+        final ExternalMovement.ExternalMovementBuilder builder = ExternalMovement.builder()
+            .movementDate(movementTime.toLocalDate())
+            .movementTime(movementTime)
+            .movementType(movementTypeRepository.findById(TAP).orElseThrow(EntityNotFoundException.withMessage(format("No %s movement type found", MovementType.TAP))))
+            .movementReason(movementReason)
+            .movementDirection(MovementDirection.IN)
+            .fromAgency(latestExternalMovement.getToAgency())
```
Comment (bell-pa) [inline src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java:891]: I think we should add .fromCity(latestExternalMovement.getToCity())

File: src/test/resources/sql/temporaryAbsenceArrival_init.sql [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                                    credited_work_hour, agreed_travel_hour, unpaid_work_supervisor,
+                                    unpaid_work_behaviour, unpaid_work_action, sick_note_received_date,
+                                    sick_note_expiry_date, court_event_result, unexcused_absence_flag,
+                                    create_user_id, modify_user_id, create_datetime, modify_datetime,
+                                    escort_code, confirm_flag, direction_code, to_city_code, from_city_code,
+                                    off_prgref_id, in_time, out_time, performance_code, temp_abs_sch_id,
+                                    reference_id, transport_code, application_time, to_country_code,
+                                    oj_location_code, contact_person_name, to_address_owner_class,
+                                    to_address_id, return_date, return_time)
+VALUES (456944515, 1176156, TO_DATE('2021-12-14', 'YYYY-MM-DD'),
```
[the remark this answers is not in the document]
Comment (michalnawrockidj) [reply src/test/resources/sql/temporaryAbsenceArrival_init.sql]: fixed

File: src/test/java/uk/gov/justice/hmpps/prison/service/TemporaryAbsenceArrivalServiceTest.java (hunk 1 above)
Comment (michalnawrockidj) [reply src/test/java/uk/gov/justice/hmpps/prison/service/TemporaryAbsenceArrivalServiceTest.java]: updated

File: src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java (hunk 2 above)
Comment (michalnawrockidj) [reply src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java:891]: done

Comment (michalnawrockidj) [reply src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java:891]: done

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**